### PR TITLE
feat(md-floating-modal): Minimized floating modal

### PR DESCRIPTION
**Pull request by mokSingh**
Minimize floating modal 

## Description
Screen Shot 2021-06-01 at 11 06 41 AM
Screen Shot 2021-06-01 at 11 06 35 AM

## Screenshots:
**Before** (If applicable):

**After**:

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: web-components/src/components/floating-modal/scss/floating.scss [hunk 1]
```diff
@@ -26,18 +33,94 @@
...
+  display: flex;
+  flex-direction: column;
+  max-height: 100%;
+  max-width: 100%;
+  position: absolute;
+  touch-action: none;
+  user-select: none;
+  z-index: 1040;
+
+  &.hide {
```
Comment (merajcal) [inline web-components/src/components/floating-modal/scss/floating.scss:70]: there are duplicate styles? is there a way to reuse?

**Review by merajcal — CHANGES_REQUESTED** [review]

File: web-components/src/components/floating-modal/FloatingMinimizedModal.ts [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+export namespace FloatingMinimizedModal {
+  @customElementWithCheck("md-floating-modal-minimized")
+  export class ELEMENT extends LitElement {
+    @property({ type: String }) heading = "";
+    @property({ type: String }) label = "";
+    @property({ type: Boolean, reflect: true }) show = false;
+    @property({ type: String, attribute: "close-aria-label" }) closeAriaLabel = "Close Modal";
+    @property({ type: Boolean, reflect: true }) minimize = false;
+    
+    @property({type: Object}) location: {
```
[the remark this answers is not in the document]
Comment (mokSingh) [reply web-components/src/components/floating-modal/FloatingMinimizedModal.ts]: Done

File: web-components/src/components/floating-modal/FloatingMinimizedModal.ts [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+      }
+      this.dragOccured = false;
+    }
+
+
+    private dragMoveListener = (event: Interact.InteractEvent) => {
+     
+      const { target, dx, dy } = event;
+     
+      const x = (parseFloat(target.getAttribute("data-x") as string) || 0) + dx + (this.applyInitialPosition && this.location ? Number(this.location?.x) : 0);
```
[the remark this answers is not in the document]
Comment (mokSingh) [reply web-components/src/components/floating-modal/FloatingMinimizedModal.ts]: Done

File: web-components/src/components/floating-modal/FloatingModal.ts [hunk 4]
```diff
@@ -48,13 +67,20 @@ export namespace FloatingModal {
...
+          this.applyInitialPosition = true;
           this.setContainerRect();
           this.setInteractInstance();
         } else {
           this.cleanContainerStyles();
           this.destroyInteractInstance();
         }
       }
+      if (this.container &&changedProperties.has("location") && !changedProperties.has("show")) {
+        if(Number(this.container?.getAttribute("data-x")) !== this.location?.x  || Number(this.container?.getAttribute("data-y")) !==this.location?.y) {
```
[the remark this answers is not in the document]
Comment (mokSingh) [reply web-components/src/components/floating-modal/FloatingModal.ts]: Done

File: web-components/src/components/floating-modal/FloatingModal.ts [hunk 5]
```diff
@@ -126,10 +162,29 @@ export namespace FloatingModal {
       );
     }
 
+    handleMinimize(event: MouseEvent) {
+      if(this.minimizable) {
+        if(!this.dragOccured) {
+          this.minimize = !this.minimize;
+          this.dispatchEvent(
+              new CustomEvent("floating-modal-minimize", {
```
[the remark this answers is not in the document]
Comment (mokSingh) [reply web-components/src/components/floating-modal/FloatingModal.ts:183]: a bit contextual properties inside the function. Have made it more clean within class

File: web-components/src/components/floating-modal/FloatingModal.ts [hunk 6]
```diff
@@ -150,13 +205,15 @@ export namespace FloatingModal {
 
     private dragMoveListener = (event: Interact.InteractEvent) => {
       const { target, dx, dy } = event;
-      const x = (parseFloat(target.getAttribute("data-x") as string) || 0) + dx;
-      const y = (parseFloat(target.getAttribute("data-y") as string) || 0) + dy;
+      const x = (parseFloat(target.getAttribute("data-x") as string) || 0) + dx + (this.applyInitialPosition && this.location ? Number(this.location?.x) : 0);
```
[the remark this answers is not in the document]
Comment (mokSingh) [reply web-components/src/components/floating-modal/FloatingModal.ts]: Extracted within class in cleaner way

File: web-components/src/components/floating-modal/scss/floating.scss (hunk 1 above)
Comment (mokSingh) [reply web-components/src/components/floating-modal/scss/floating.scss]: Done, remvoed all duplicated styles

**Review by merajcal — APPROVED** [review]

**Pull request merged**